### DISCUSSION TITLE
I need help to create a new bind to toggle some of the commands!

### DISCUSSION THREAD

**Discussion by Dexteriority** [winstxnhdw/lc-hax#302] [Q&A] [answered]
I had read about the toggle clear vision discussion, i also understand just a little of coding.
But when i open the InputListener file from Script/Core like you said in other discussion, im confused of what changes should i make to create a new input bind for spesific command??

For example in **public class InputListener : MonoBehaviour {** Tab, we have **public static event Action? onMiddleButtonPress;** .

And for **(Func<bool>, Action)[] InputActions { get; } = [** Tab, we have **(() => Mouse.current.middleButton.wasPressedThisFrame, () => InputListener.onMiddleButtonPress?.Invoke()),**.

But in all those _MiddleButtonPress_ i dont see any code saying "Trigger"? Because M3 is used as Trigger bind right?
I really need help about this:((
P.S I dont learn it to make another copy of this lchax like other cheat menus dont worry :)

**Comment by winstxnhdw** [comment]
Think of every key in that file as a global bind. For example, if you change M3 to  Backspace. Now every bind in this mod that used to use M3, is now Backspace. I hope that explanation helps.

**Comment by winstxnhdw** [comment]
Creating a new toggle is much trickier, because you now have to go into the implementation of the mod you want to toggle and create an if-guard that will block the command from running. You can see the ESPMod for an example of how we do that.

**Reply by Dexteriority** [reply]
so it's not possible to create the one like code above? since modifying is a lot easier to learn than creating xD

**Reply by winstxnhdw** [reply]
I am not sure which code you are specifying. The only code you mentioned doesn’t make sense in this context.

**Reply by Dexteriority** [reply]
Im deeply apologize for the confusion, I'm not trying to change the example, I wanted to bind InfiniteGrab,NoWeight, and Godmod^^~

**Reply by winstxnhdw** [reply]
I see. Yeah, you’ll have to add a flag to each implementation. It would be a bit tricky with zero experience but if you’ve done some basic programming before, this should be trivial.

**Reply by Dexteriority** [reply]
I can't call myself zero experience but I also didn't know what to type to create new flag;(

**Reply by winstxnhdw** [reply, marked as answer]
Here's an example implementation for `WeightMod`

```cs
using System.Collections;
using GameNetcodeStuff;
using UnityEngine;
using Hax;

public sealed class WeightMod : MonoBehaviour
{
    bool EnableWeightMod { get; set; } = false;

    void OnEnable()
    {
        InputListener.onMiddleButtonPress += this.ToggleWeight;
    }

    void OnDisable()
    {
        InputListener.onMiddleButtonPress -= this.ToggleWeight;
    }

    void ToggleWeight() => this.EnableWeightMod = !this.EnableWeightMod;

    IEnumerator SetWeight(object[] args)
    {
        WaitForEndOfFrame waitForEndOfFrame = new();
        WaitForSeconds waitForOneSecond = new(1.0f);

        while (true)
        {
            if (!this.EnableWeightMod)
            {
                yield return waitForEndOfFrame;
                continue;
            }

            if (Helper.LocalPlayer is not PlayerControllerB player)
            {
                yield return waitForEndOfFrame;
                continue;
            }

            player.carryWeight = 1.0f;
            yield return waitForOneSecond;
        }
    }

    void Start()
    {
        _ = this.StartResilientCoroutine(this.SetWeight);
    }
}
```

**Reply by Dexteriority** [reply]
omg thank you very much!! 
i will be use and modifying this to create another like Stamina,Godmode,etc.
Again thank you very much!

**Reply by Dexteriority** [reply]
eh why is it errors XD

**Reply by winstxnhdw** [reply]
What's the error?

**Reply by Dexteriority** [reply]
I should replace those codes in my Module folder right?
My errors are:
![image](https://github.com/winstxnhdw/lc-hax/assets/157716300/8e52aa20-b311-48a7-aae8-94272627862a)

**Reply by winstxnhdw** [reply]
It's just missing a semi-colon. I've edited the script. Should be correct now.

**Reply by Dexteriority** [reply]
now it says the another error on colum: 12, 9 and 17, 9 ? what should i place there? I tried removing the line and adding semi-colon(ik im dumb) but still ://
I dont really know if the spam red message below it are necessary to get in the shot xd

![image](https://github.com/winstxnhdw/lc-hax/assets/157716300/9f2a92d4-66f4-458c-83ca-988b7e8ccb45)

**Reply by winstxnhdw** [reply]
Oops. It’s because I am using the wrong input listener. Lemme change it.

**Reply by Dexteriority** [reply]
Im excited to learn

**Reply by Dexteriority** [reply]
It works! But i got this glitch where i got myself to 70 - 90 lbs, toggled off the WeightMod and drop all of the items, i got speedhack xD
I understand how this occurs, i can play around with it wont be so annoyingXDXD.

Anyway the codes work perfectly fine, if i may ask another question, is _ToggleWeight_ in the code are new code? so that we can change it to toggle whatever we wanted based from this code? 
Again, Thank you very much for your hard work!!!

**Reply by winstxnhdw** [reply]
I am not sure what you mean.

**Reply by Dexteriority** [reply]
I figured out myself that i can modify the bind base you created XD
Im satisfied. great work!
